zebra-state/src/service/finalized_state/tests/vectors.rs

```diff
@@ -73,21 +80,8 @@ fn sprout_note_commitment_tree_serialization() {
     // The purpose of this test is to make sure the serialization format does
```

Comment:
Just turning this into a checklist, so I can make sure the tests cover both serialization formats:
(Or we can work out which tests aren't actually needed.)

Currently, these tests check these structs are equal:
- [ ] commitments -> tree struct
- [ ] commitments -> tree struct -> serialize -> deserialize -> tree struct
And these serialized formats are equal:
- [ ] fixed serialized test vector
- [ ] commitments -> tree struct -> serialize
- [ ] commitments -> tree struct -> serialize -> deserialize -> tree struct -> serialize

TODO:
- [ ] apply these tests to the new tree structs, and update the serialization format
- [ ] do we need to test the old format as well, or do the tests below cover it?

TODO: test that old and new serializations produce the same format:
Tree roots built from the same commitments should match:
- [ ] commitments -> old tree struct -> new tree struct -> un-cached root
- [ ] commitments -> new tree struct -> un-cached root
Even when serialized and deserialized:
- [ ] commitments -> old tree struct -> old serialize -> old deserialize -> old tree struct ->  new tree struct -> un-cached root
- [ ] commitments -> new tree struct -> new serialize -> new deserialize -> new tree struct -> un-cached root
- [ ] commitments -> new tree struct -> un-cached root

Reply:
I think we're missing these tests:
- using the old tree struct with the same commitments produces the same root
- using the old serialization format to create a new tree struct produces the same root

We can test this two different ways:
Keep the old serialization code, and do these tests:
- commitments -> old tree struct -> new tree struct -> un-cached root
- commitments -> old tree struct -> old serialize -> old deserialize -> old tree struct -> new tree struct -> un-cached root
- (existing new root)

Keep the old serialzation format and root as fixed test vectors, and do these tests:
- old serialization format -> new tree struct -> un-cached root
- old root fixed test vector
- (existing new root)

@oxarbitrage which way did you want to go?

Reply:
I don't think we will need so many tests now that we are keeping the old serialization.

Reply:
Yep, if we're only using one format, then it's enough to check that the new struct roots match the old struct roots.

Something like `assert_eq!(tree.recalculate_root(), tree.cached_root().expect("cached root was serialized"));`

Reply:
Added those at https://github.com/ZcashFoundation/zebra/pull/7053/commits/0b494db159eacd7b388f268110404c7997031999